onnxruntime/core/providers/cpu/nn/roi_pool.cc

```diff
@@ -61,10 +56,10 @@ Status RoiPool<float>::Compute(OpKernelContext* context) const {
           // Compute pooling region for this output unit:
           //  start (included) = floor(ph * roi_height / pooled_height_)
           //  end (excluded) = ceil((ph + 1) * roi_height / pooled_height_)
-          int hstart = static_cast<int>(std::floor(static_cast<float>(ph) * bin_size_h));
-          int wstart = static_cast<int>(std::floor(static_cast<float>(pw) * bin_size_w));
-          int hend = static_cast<int>(std::ceil(static_cast<float>(ph + 1) * bin_size_h));
-          int wend = static_cast<int>(std::ceil(static_cast<float>(pw + 1) * bin_size_w));
+          int hstart = (ph * roi_height) / static_cast<int>(pooled_height_);
+          int wstart = (pw * roi_width) / static_cast<int>(pooled_width_);
+          int hend = ((ph + 1) * roi_height + static_cast<int>(pooled_height_) - 1) / static_cast<int>(pooled_height_);
+          int wend = ((ph + 1) * roi_width + static_cast<int>(pooled_width_) - 1) / static_cast<int>(pooled_width_);
```

Comment:
Is it possible to add a unit test to validate the new logic so that it doesn't get accidentally broken in the future?

Reply:
The old version wasn't really broken, it's just that this version is more precise and stable. In 99% of the cases it will give the same results. I can't think of a test that would have made the old implementation fail but the new implementation pass, at least on the CPU.